File: src/test/java/com/agiledeveloper/Triple.java
package com.agiledeveloper;

import java.util.*;
import java.util.stream.*;
import static java.util.stream.Collectors.*;

class Triple {
  public static String getTriple(int m, int n) {   
    int a = m * m - n * n;
    int b = 2 * m * n;
    int c = m * m + n * n;
    
    return String.format("%d %d %d", a, b, c);
  }
  
  public static List<String> computeTriples(int numberOfValues) {

    return IntStream.range(2, 2 + numberOfValues)
      .boxed()
      .flatMap(m -> 
        IntStream.range(1, m)
          .mapToObj(n -> getTriple(m, n))
      )
      .limit(numberOfValues)
      .collect(toList());
 }  
}

File: src/test/scala/com/agiledeveloper/Triple.scala
package com.agiledeveloper

object Triple {
  def getTriple(m: Int, n: Int): String = {   
    val a = m * m - n * n
    val b = 2 * m * n
    val c = m * m + n * n
    
    s"$a $b $c"
  }
  
  def computeTriples(numberOfValues: Int): List[String] = {
    val triples = for {
      m <- (2 to (numberOfValues + 1))
      n <- 1 until m
    } yield getTriple(m, n)

    triples.take(numberOfValues).toList
  }
}